### PR TITLE
Add Logging support for the Infection

### PR DESCRIPTION
**Pull request by BlairArchibald**
This allows easier handling of testing by tracking when patients became symptomatic, and also allows tracking of, for example, the R0 value.

### CODE REVIEW

File: src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java [hunk 1]
```diff
@@ -109,15 +114,15 @@ else if(symptomaticCase) {
...
-    	if(symptomaticCase) status = this.stepInfectionSymptomatic();
-    	else if(!symptomaticCase) status = this.stepInfectionAsymptomatic();
+    	if(symptomaticCase) status = this.stepInfectionSymptomatic(t);
+    	else if(!symptomaticCase) status = this.stepInfectionAsymptomatic(t);
     	return status;
     }
     
     // Cycle through the infection for that timestep
-    public CStatus stepInfectionAsymptomatic() {
+    public CStatus stepInfectionAsymptomatic(Time t) {
```
Comment (bobtaylor12) [inline src/main/java/uk/co/ramp/covid/simulation/covid/Covid.java:125]: t is not used in this method

File: src/main/java/uk/co/ramp/covid/simulation/covid/InfectionLog.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+package uk.co.ramp.covid.simulation.covid;
+
+import uk.co.ramp.covid.simulation.Time;
+import uk.co.ramp.covid.simulation.population.Person;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/** InfectionLog matains information about a particular COVID case, such as when it became
```
Comment (bobtaylor12) [inline src/main/java/uk/co/ramp/covid/simulation/covid/InfectionLog.java:9]: typo

**Comment by johnnonweiler** [comment]
I think the column in the output marked "R0" should probably be "R" (because it is the value after some people have become infected, and can then not be infected again)?

**Comment by BlairArchibald** [comment]
> I think the column in the output marked "R0" should probably be "R" (because it is the value after some people have become infected, and can then not be infected again)?

I'm not too sure on the differences between R0 and effective R - @paulbessell what should this be?

**Comment by paulbessell** [comment]
> > I think the column in the output marked "R0" should probably be "R" (because it is the value after some people have become infected, and can then not be infected again)?
> 
> I'm not too sure on the differences between R0 and effective R - @paulbessell what should this be?

Technially it is R. R0 is when the population is fully susceptible, so is a bit of a paper concept. However, it is largely a semantic issue

**Comment by BlairArchibald** [comment]
Great! Thanks @paulbessell  and @johnnonweiler

**Pull request merged**